consumer/git-publisher/src/main/java/org/arquillian/pact/consumer/publisher/git/GitPactPublisher.java

```diff
@@ -0,0 +1,316 @@
+package org.arquillian.pact.consumer.publisher.git;
+
+import org.arquillian.pact.common.configuration.PactRunnerExpressionParser;
+import org.arquillian.pact.common.git.GitOperations;
+import org.arquillian.pact.consumer.spi.publisher.PactPublisher;
+import org.eclipse.jgit.api.Git;
+import org.eclipse.jgit.api.PullResult;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardCopyOption;
+import java.util.Map;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.stream.Stream;
+
+import static org.arquillian.pact.common.configuration.HomeResolver.resolveHomeDirectory;
+
+public class GitPactPublisher implements PactPublisher {
+
+    static final String URL = "url";
+    static final String USERNAME = "username";
+    static final String PASSWORD = "password";
+    static final String PASSPHRASE = "passphrase";
+    static final String REMOTE = "remote";
+    static final String KEY = "key";
+    static final String REPOSITORY = "repository";
+    static final String PACT_DIRECTORY = "pactDirectory";
+    static final String TAG = "tag";
+    static final String BRANCH = "branch";
+    static final String COMMENT = "comment";
+    static final String EMAIL = "email";
+
+    private static final Logger logger = Logger.getLogger(GitPactPublisher.class.getName());
+
+    private Map<String, Object> configuration;
+    GitOperations gitOperations;
+
+
+    public GitPactPublisher() {
+        this.gitOperations = new GitOperations();
+    }
+
+    @Override
+    public void store(Path pactsLocation) {
+
+        Git git = null;
+        try {
+            if (isSet(REPOSITORY, String.class, this.configuration)) {
+
+                Path repository = Paths.get(getResolvedValue((String) this.configuration.get(REPOSITORY)));
+
+                if (this.gitOperations.isValidGitRepository(repository)) {
+
+                    git = this.gitOperations.openGitRepository(repository);
```

Comment:
My understanding of `GitOperations` (which we actually could give better name) is that we provide an adapter with some high level commands and encapsulate direct operations performed on `org.eclipse.jgit.api.Git`. Here we are mixing both. Can we improve it?

This could also be applied to conditional operations user/pwd or passphare.